DEVELOPER.md

```diff
@@ -1,4 +1,193 @@
+as easily switching across versions, configurations and dependencies. It
+has support for running Kong in Hybrid (CP/DP) mode, testing migrations,
+running a Kong cluster, among other [features](https://github.com/Kong/gojira/blob/master/doc/manual.md).
+
+#### Kong Pongo
+
+[Pongo](https://github.com/Kong/kong-pongo) is another CLI like Gojira,
+but specific for plugin development. It is docker-compose based and will
+create local test environments including all dependencies. Core features
+are running tests, integrated linter, config initialization, CI support,
+and custom dependencies.
+
+#### Kong Plugin Template
+
+The [plugin template](https://github.com/Kong/kong-plugin) provides a basic
+plugin and is considered a best-practices plugin repository. When writing
+custom plugins we strongly suggest you start by using this repository as a
+starting point. It contains the proper file structures, configuration files,
+and CI setup to get up and running quickly. This repository seamlessly
+integrates with [Pongo](https://github.com/Kong/kong-pongo).
+
+#### Vagrant
+
+You can use a Vagrant box running Kong and Postgres that you can find at
+[Kong/kong-vagrant](https://github.com/Kong/kong-vagrant).
+
+#### Source Install
+
+Kong mostly is an OpenResty application made of Lua source files, but also
+requires some additional third-party dependencies. We recommend installing
+those by following the source install instructions at
+https://docs.konghq.com/install/source/.
+
+Instead of following the second step (Install Kong), clone this repository
+and install the latest Lua sources instead of the currently released ones:
+
+```shell
+$ git clone https://github.com/Kong/kong
+$ cd kong/
+
+# you might want to switch to the development branch. See CONTRIBUTING.md
+$ git checkout master
+
+# install the Lua sources
+$ luarocks make
+```
+
+#### Running for development
+
+Check out the [development section](https://github.com/Kong/kong/blob/master/kong.conf.default#L244)
+of the default configuration file for properties to tweak in order to ease
+the development process for Kong.
+
+Modifying the [`lua_package_path`](https://github.com/openresty/lua-nginx-module#lua_package_path)
+and [`lua_package_cpath`](https://github.com/openresty/lua-nginx-module#lua_package_cpath)
+directives will allow Kong to find your custom plugin's source code wherever it
+might be in your system.
+
+#### Tests
+
+Install the development dependencies ([busted], [luacheck]) with:
+
+```shell
+$ make dev
+```
+
+Kong relies on three test suites using the [busted] testing library:
+
+* Unit tests
+* Integration tests, which require Postgres and Cassandra to be up and running
+* Plugins tests, which require Postgres to be running
+
+The first can simply be run after installing busted and running:
+
+```
+$ make test
+```
+
+However, the integration and plugins tests will spawn a Kong instance and
+perform their tests against it. As so, consult/edit the `spec/kong_tests.conf`
```

Comment:
```suggestion
perform their tests against it. Because these test suites perform their tests against the Kong instance, you may need to edit the `spec/kong_tests.conf`
```